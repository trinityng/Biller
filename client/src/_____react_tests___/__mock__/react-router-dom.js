import React from 'react';
// Just render plain div with its children
const rrd = require('react-router-dom');

rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;