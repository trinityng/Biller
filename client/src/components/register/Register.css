.register-form {
  max-width: 500px;
  margin: 0 auto;
}

.register-inputs {
  max-width: 470px;
}


.register-button {
  width: 100%;
}

.register-input {
  color: "rgba(0,0,0,.25)";
}