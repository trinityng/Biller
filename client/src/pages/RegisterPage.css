.register-box {
    display: block;
    position: relative;
    border-radius: 4px;
    border: 5px #ddd solid;
    background-color: rgb(255, 255, 255);
}

.register-center {
    display: 'flex';
    width: 600px;
    overflow-y: auto;
    margin: 0 auto;
}