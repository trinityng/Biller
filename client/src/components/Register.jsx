import React from 'react';
import { Button, Input, Form, Row, Col } from 'antd';
import { Link } from 'react-router-dom';
import { UserOutlined, LockOutlined } from '@ant-design/icons';

const Register = props => {
  const onFinish = values => {
    props.onSubmission(values);
  };

  return (
    <Form name="register" className="input-form" onFinish={onFinish}>
      <Link to="/" className="register-sign-up">
        <img src={require('../assets/logo.png')} alt="logo" />
      </Link>
      <h2>Register</h2>
      <Form.Item
        name="email"
        hasFeedback
        rules={[{ required: true, message: 'Please input your email!' }]}
      >
        <Input
          prefix={<UserOutlined className="site-form-item-icon" />}
          type="email"
          placeholder="Email"
        />
      </Form.Item>

      <Form.Item
        name="password"
        hasFeedback
        rules={[
          { required: true, message: 'Please input your Password!' },
          { min: 8, whitespace: true, message: 'Min length: 8' },
        ]}
      >
        <Input.Password
          prefix={<LockOutlined className="site-form-item-icon" />}
          type="password"
          placeholder="Password"
        />
      </Form.Item>

      <Form.Item
        name="confirm"
        dependencies={['password']}
        hasFeedback
        rules={[
          { required: true, message: 'Please confirm your password!' },
          ({ getFieldValue }) => ({
            validator(rule, value) {
              if (!value || getFieldValue('password') === value)
                return Promise.resolve();

              return Promise.reject(
                new Error('The two passwords that you entered do not match!')
              );
            },
          }),
        ]}
      >
        <Input.Password
          prefix={<LockOutlined className="site-form-item-icon" />}
          type="password"
          placeholder="Confirm Password"
        />
      </Form.Item>

      <Form.Item>
        <Row>
          <Col span={12}>
            <Form.Item
              name="firstname"
              hasFeedback
              rules={[
                { required: true, message: 'Please enter your first name!' },
              ]}
            >
              <Input placeholder="First Name" />
            </Form.Item>
          </Col>
          <Col span={12}>
            <Form.Item
              name="lastname"
              hasFeedback
              rules={[
                { required: true, message: 'Please enter your last name!' },
              ]}
            >
              <Input placeholder="Last Name" />
            </Form.Item>
          </Col>
        </Row>
      </Form.Item>

      <Form.Item>
        <Button type="primary" htmlType="submit" className="register-button">
          Sign up
        </Button>
      </Form.Item>
    </Form>
  );
};
export default Register;
