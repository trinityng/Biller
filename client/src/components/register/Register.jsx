import React from "react";
import { Button, Input, Form, Row, Col } from "antd";
import "antd/dist/antd.css";

import "./Register.css";

const Register = props => {
  const onFinish = values => {
    console.log('Sent values:', values);
    props.onClick(values)
  };

  return (
    <Form
      name="register"
      className="register-form"
      onFinish={onFinish}
    >
      <Form.Item/>
      <Form.Item
        name="username"
        hasFeedback
        rules={[ () => ({
            validator(rule, value) {
              if (/.+@.+\..+/.test(value)) {
                return Promise.resolve();
              }
              return Promise.reject('Your email is not valid!');
            },
          }),
        ]}
      >
         <Input type = "email"  placeholder="Email Account" />
      </Form.Item>

      <Form.Item
        name="password"
        hasFeedback
        rules={[{ required: true, message: 'You require a password!' }, 
        () => ({
          validator(rule, value) {
            if (value.length >= 6) {
              return Promise.resolve();
            }
            return Promise.reject('Your password must be 6 or more characters!');
          },
        }),]}
      >
        <Input type = "password" placeholder="Password" />
      </Form.Item>

      <Form.Item
        name="confirm"
        dependencies={['password']}
        hasFeedback
        rules={[ {required: true, message: 'Please confirm your password!' },
          ({ getFieldValue }) => ({
            validator(rule, value) {
              if (!value || getFieldValue('password') === value) {
                return Promise.resolve();
              }
              return Promise.reject('The two passwords that you entered do not match!');
            },
          }),
        ]}
      >
        <Input type = "password" placeholder="Confirm Password" />
      </Form.Item>

      <Form.Item>
        <Row>
          <Col span={12}>
          <Form.Item
            name="firstname"
            hasFeedback
            rules={[{ required: true, message: 'Please enter your first name!' }]}
          >
             <Input placeholder="First Name" />
          </Form.Item>
          </Col>
          <Col span={12}> 
          <Form.Item
            name="lastname"
            hasFeedback
            rules={[{ required: true, message: 'Please enter your last name!' }]}
          > 
            <Input placeholder="Last Name" />
          </Form.Item>
          </Col>
        </Row>
      </Form.Item>

      <Form.Item/>

      <Form.Item>
        <Button
          type="primary"
          htmlType="submit"
          className="register-button"
        > 
          Sign up
        </Button>
      </Form.Item>
    </Form>
  );
}
export default Register;
