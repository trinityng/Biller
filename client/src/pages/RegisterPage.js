import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Redirect, withRouter } from 'react-router-dom';
import { Layout } from "antd";

import { register } from '../api/register.api';
import Register from '../components/register/Register';
import history from '../router/History';

import "./Pages.css"

class RegisterPage extends Component {
  onSubmission = e => {
    console.log("Recieved values: ", e);

    this.props.register({
      email: e.username,
      password: e.password,
      firstname: e.firstname,
      lastname: e.lastname
    })

    if(this.props.success) {
      history.push('/setup');  
    }
  };

  render() {
    const user = this.props.user || {};
    const { token } = user;

    return token ? (
      <Redirect to="/" user={user} />
    ) : (
      <Register 
        onSubmission={e => this.onSubmission(e)}
      />  
    );
  }
}

// Store
function mapStateToProps(state) {
  return {
    user: state.user,
  };
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators({ register }, dispatch);
}

export default connect(
  mapStateToProps,
  matchDispatchToProps
)(withRouter(RegisterPage));
