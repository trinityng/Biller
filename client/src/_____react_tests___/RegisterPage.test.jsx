import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import {mount, shallow, configure } from 'enzyme';
import { MemoryRouter, Router } from "react-router-dom";
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

import RegisterPage from '../pages/RegisterPage';
import Register from '../components/Register';


Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

configure({adapter: new Adapter()});

describe('register page / component', () => {
  let initialState = {};
  let store;

  beforeEach(() => {
    store = mockStore(initialState)
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/register']}>
          <RegisterPage />
        </MemoryRouter>,
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe("when user is logged in", () => {
    it('should redirect the user to dashboard', () => {
     
    });

    it('should redirect the user to setup page', () => {
     
    });
  });
});