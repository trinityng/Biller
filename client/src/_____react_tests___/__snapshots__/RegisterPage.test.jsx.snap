// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register page / component should render correctly 1`] = `ReactWrapper {}`;
